### PR TITLE
Full Auto Fire Fix

### PR DESCRIPTION
**Pull request by Geevies**
* Fixed there not being a delay when full auto firing a gun.

### CODE REVIEW

**Comment by Geevies** [comment]
!bugfix

**Comment by Geevies** [comment]
!review

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -297,11 +299,13 @@
...
 
 	if(world.time < next_fire_time)
-		if (world.time % 3) //to prevent spam
+		if(world.time % 3 && !can_autofire) //to prevent spam
 			to_chat(user, SPAN_WARNING("\The [src] is not ready to fire again!"))
 		return FALSE
 
 	var/shoot_time = (burst - 1) * burst_delay
+	if(can_autofire)
+		shoot_time = DEFAULT_AUTOFIRE_BASE_DELAY * burst_delay
```
Comment (Arrow768) [inline code/modules/projectiles/gun.dm:308]: That multiplication does not make sense.
Why would you multiply the DEFAULT_AUTOFIRE_BASE_DELAY (1) with the burst_delay.
A multiplication with 1 does not change the value in any way.

**Review by Arrow768 — DISMISSED** [review]

**Comment by Geevies** [comment]
!review

**Review by skull132 — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

**Pull request merged**